service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java

```diff
@@ -397,8 +400,33 @@ public PipelineRun markPipelineRunSuccessAndWriteOutputs(
   // methods to interact with and format pipeline run outputs
 
   /**
-   * Extract the pipeline outputs from a pipelineRun object and return an ApiPipelineRunOutputs
-   * object with the outputs.
+   * Extract pipeline outputs from a Rawls entity object, converting wdlVariableName (typically
+   * snake_case) to outputName (camelCase). Throw an error if any outputs are missing from the
+   * entity or empty.
+   *
+   * @param pipelineOutputDefinitions
+   * @param entity
+   * @return a map of pipeline outputs
+   */
+  public Map<String, String> extractPipelineOutputsFromEntity(
```

Comment:
just a random thought i had about the future.  I wonder if it would make sense to put these input/output focused function into its own service if this one starts getting too bloated

Reply:
I agree, it felt a little odd to put this into PipelineRuns service. I'd gladly move some of these methods into their own service/utility in future